src/ucp/wireup/select.c

```diff
@@ -1872,20 +1877,27 @@ ucp_wireup_add_rma_bw_lanes(const ucp_wireup_select_params_t *select_params,
 
         /* Add lanes that can access the memory by short operations */
         added_lanes = 0;
-        UCS_BITMAP_CLEAR(&tl_bitmap);
-
-        ucs_memory_type_for_each(mem_type) {
-            UCP_CONTEXT_MEM_CAP_TLS(context, mem_type, reg_mem_types,
-                                    mem_type_tl_bitmap);
-
-            bw_info.criteria.reg_mem_types = UCS_BIT(mem_type);
-            added_lanes                   += ucp_wireup_add_bw_lanes(
-                                               select_params, &bw_info,
-                                               UCP_TL_BITMAP_AND_NOT(
-                                                 mem_type_tl_bitmap, tl_bitmap),
-                                               UCP_NULL_LANE, select_ctx);
 
-            UCS_BITMAP_OR_INPLACE(&tl_bitmap, mem_type_tl_bitmap);
+        while (lanes_iter--) {
+            added_lanes_prev = added_lanes;
+            UCS_BITMAP_CLEAR(&tl_bitmap);
+            ucs_memory_type_for_each(mem_type) {
+                UCP_CONTEXT_MEM_CAP_TLS(context, mem_type, reg_mem_types,
+                                        mem_type_tl_bitmap);
+
+                bw_info.criteria.reg_mem_types = UCS_BIT(mem_type);
+                added_lanes                   += ucp_wireup_add_bw_lanes(
```

Comment:
due to `dev_count` being declared inside `ucp_wireup_add_bw_lanes()`, when now calling multiple times for a given memtype, i think we could end-up failing to track local and remote max usage count against `dev_num_path` properly?

Reply:
That's right. Maybe we should initialize define `dev_count` in add_rma_bw (and other functions making call to add_bw_lanes) and pass it to add_bw_lanes.